contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -389,7 +391,10 @@ func (s *SchedulerServer) prepareExecutorInfo(hks hyperkube.Interface) (*mesos.E
 	} else if !hks.FindServer(hyperkube.CommandMinion) {
 		return nil, fmt.Errorf("either run this scheduler via km or else --executor-path is required")
 	} else {
-		if strings.Index(s.kmPath, "://") > 0 {
+		if s.kmLocalPath != "" { // The default value of kmLocalPath is "".
+			// TODO(k82): Check whether `kmLocalPath` start with `file://` or `/`.
```

Comment:
What do you think about making this a subset of the `://` case?
```go
} else if strings.Index(s.kmPath, "://") > 0 {
  if strings.startsWith(s.kmPath, "file://") {
    // ... don't add this as a URI because it's local to the agent; parse the basename of kmPath and use that for ci.Valkue
    // (new bits)
  } else {
    // ... assume this is something to be downloaded; append a URI and assume that `km` will land in the sandbox.
    // (exiting bits)
  }
} else if ...
```